src/routes/patreon.ts

```diff
@@ -1,33 +1,35 @@
 import express from 'express';
 import {
+  MultiplePatreonsForUserError,
+  MultipleUsersForPatreonError,
   PatreonAgent,
-  PatronPublicDetails,
+  PatronDetails,
 } from '../clients/patreon/patreonAgent';
 import { PatreonController } from '../clients/patreon/patreonController';
 
 const router = express.Router();
 const patreonAgent = new PatreonAgent(new PatreonController());
 
 router.get('/oauth/redirect', async (req, res) => {
+  if (!req.query.code) {
+    // If a user does not grant access to their Patreon account on redirect
+    return res.redirect(
+      // @ts-ignore
+      `${req.session.postPatreonRedirectUrl}?error=We were unable to retrieve your Patreon details.`,
+    );
+  }
+
   if (
     !req.query.state ||
     // @ts-ignore
     req.query.state.toString() !== req.session.patreonAuthState
   ) {
-    // TODO-kev: Figure a way to handle this. Redirect to homepage? Swal? Should not be based on redirectUrl
-    console.error('CSRF detected.');
-    return res.status(403).redirect(
-      // @ts-ignore
-      `${req.session.postPatreonRedirectUrl}`,
-    );
-  }
-
-  if (!req.query.code) {
-    // If a user does not grant access to their Patreon account on redirect
-    return res.redirect(
-      // @ts-ignore
-      `${req.session.postPatreonRedirectUrl}?error=We were unable to retrieve details to link a Patreon account.`,
+    // @ts-ignore
+    req.flash(
+      'error',
+      'Something went wrong. Please contact an admin if you see this.',
```

Comment:
Let's have a log line here. Even logging this error message is better than nothing. Will help us debug. if someone hits it.